amazon-s3-quickstart fails with Quarkus main

amazon-s3-quickstart fails with Quarkus main, I see the fail in our Jenkins daily runs for last 6 days.

https://github.com/quarkusio/quarkus-quickstarts/actions is red for quite some time (2 weeks) too.

```
org.acme.s3.S3ResourcesTest.testResource(String)[2]

1 expectation failed.
Response body doesn't match expectation.
Expected: "Cherry"
  Actual: Cherry&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;&amp;amp#0;

	at org.acme.s3.S3ResourcesTest.lambda$testResource$1(S3ResourcesTest.java:61)
	at org.acme.s3.S3ResourcesTest.testResource(S3ResourcesTest.java:54)
```

Looks like infinispan is failing now:
```
2021-11-15T11:15:33.9382375Z [INFO] Running org.acme.infinispan.client.InfinispanGreetingResourceTest
2021-11-15T11:15:39.8712089Z [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 5.925 s <<< FAILURE! - in org.acme.infinispan.client.InfinispanGreetingResourceTest
2021-11-15T11:15:39.8724796Z [ERROR] org.acme.infinispan.client.InfinispanGreetingResourceTest.testHelloEndpoint  Time elapsed: 0.011 s  <<< ERROR!
2021-11-15T11:15:39.8733025Z java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
2021-11-15T11:15:39.8738070Z Caused by: java.lang.reflect.InvocationTargetException
2021-11-15T11:15:39.8744673Z Caused by: java.lang.RuntimeException: Error waiting for test resource future to finish.
2021-11-15T11:15:39.8752664Z Caused by: java.util.concurrent.ExecutionException: java.lang.NoClassDefFoundError: org/junit/rules/TestRule
2021-11-15T11:15:39.8759317Z Caused by: java.lang.NoClassDefFoundError: org/junit/rules/TestRule
2021-11-15T11:15:39.8765662Z Caused by: java.lang.ClassNotFoundException: org.junit.rules.TestRule
2021-11-15T11:15:39.8767264Z 
2021-11-15T11:15:39.9242736Z [INFO] 
2021-11-15T11:15:39.9244126Z [INFO] Results:
2021-11-15T11:15:39.9245077Z [INFO] 
2021-11-15T11:15:39.9245963Z [ERROR] Errors: 
2021-11-15T11:15:39.9248622Z [ERROR]   InfinispanGreetingResourceTest.testHelloEndpoint ? Runtime java.lang.reflect.I...
2021-11-15T11:15:39.9254577Z [INFO] 
2021-11-15T11:15:39.9255846Z [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
```

--- a/amazon-s3-quickstart/src/main/java/org/acme/s3/S3AsyncClientResource.java
+++ b/amazon-s3-quickstart/src/main/java/org/acme/s3/S3AsyncClientResource.java
@@ -19,9 +19,11 @@ import org.jboss.resteasy.annotations.jaxrs.PathParam;
 import org.jboss.resteasy.annotations.providers.multipart.MultipartForm;
 
 import io.smallrye.mutiny.Uni;
+import software.amazon.awssdk.core.ResponseBytes;
 import software.amazon.awssdk.core.async.AsyncRequestBody;
 import software.amazon.awssdk.core.async.AsyncResponseTransformer;
 import software.amazon.awssdk.services.s3.S3AsyncClient;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
 import software.amazon.awssdk.services.s3.model.ListObjectsRequest;
 import software.amazon.awssdk.services.s3.model.ListObjectsResponse;
 import software.amazon.awssdk.services.s3.model.S3Object;
@@ -58,15 +60,13 @@ public class S3AsyncClientResource extends CommonResource {
     @GET
     @Path("download/{objectKey}")
     @Produces(MediaType.APPLICATION_OCTET_STREAM)
-    public Uni<Response> downloadFile(@PathParam("objectKey") String objectKey) throws Exception {
-        File tempFile = tempFilePath();
-
+    public Uni<Response> downloadFile(@PathParam("objectKey") String objectKey) {
         return Uni.createFrom()
-                .completionStage(() -> s3.getObject(buildGetRequest(objectKey), AsyncResponseTransformer.toFile(tempFile)))
+                .completionStage(() -> s3.getObject(buildGetRequest(objectKey), AsyncResponseTransformer.toBytes()))
                 .onItem()
-                .transform(object -> Response.ok(tempFile)
+                .transform(object -> Response.ok(object.asUtf8String())
                         .header("Content-Disposition", "attachment;filename=" + objectKey)
-                        .header("Content-Type", object.contentType()).build());
+                        .header("Content-Type", object.response().contentType()).build());
     }
 
     @GET

--- a/amazon-s3-quickstart/src/main/java/org/acme/s3/S3SyncClientResource.java
+++ b/amazon-s3-quickstart/src/main/java/org/acme/s3/S3SyncClientResource.java
@@ -20,6 +20,7 @@ import javax.ws.rs.core.StreamingOutput;
 import org.jboss.resteasy.annotations.jaxrs.PathParam;
 import org.jboss.resteasy.annotations.providers.multipart.MultipartForm;
 
+import software.amazon.awssdk.core.ResponseBytes;
 import software.amazon.awssdk.core.sync.RequestBody;
 import software.amazon.awssdk.core.sync.ResponseTransformer;
 import software.amazon.awssdk.services.s3.S3Client;
@@ -59,12 +60,10 @@ public class S3SyncClientResource extends CommonResource {
     @Path("download/{objectKey}")
     @Produces(MediaType.APPLICATION_OCTET_STREAM)
     public Response downloadFile(@PathParam("objectKey") String objectKey) {
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        GetObjectResponse object = s3.getObject(buildGetRequest(objectKey), ResponseTransformer.toOutputStream(baos));
-
-        ResponseBuilder response = Response.ok((StreamingOutput) output -> baos.writeTo(output));
+        ResponseBytes<GetObjectResponse> objectBytes = s3.getObjectAsBytes(buildGetRequest(objectKey));
+        ResponseBuilder response = Response.ok(objectBytes.asUtf8String());
         response.header("Content-Disposition", "attachment;filename=" + objectKey);
-        response.header("Content-Type", object.contentType());
+        response.header("Content-Type", objectBytes.response().contentType());
         return response.build();
     }
 

